### PR TITLE
[*] Technical - Let Forest developers override the default url for the client app

### PR DESCRIPTION
**Pull request by arnaudbesnier**

### CODE REVIEW

**Comment by SeyZ** [comment]
In my opinion, you should not write any logic in the generated back offices for all the Lumber's users.

File: services/dumper.js [hunk 1]
```diff
@@ -130,7 +130,8 @@ function Dumper(project, config) {
       ssl: config.ssl,
       encrypt: config.ssl && config.dbDialect === 'mssql',
       dbSchema: config.dbSchema,
-      port: config.appPort
+      port: config.appPort,
+      frontendHost: config.frontendHost
```
Comment (SeyZ) [inline services/dumper.js]: See my global remark on this. For me, we should keep this file untouched.

File: services/prompter.js [hunk 2]
```diff
@@ -19,6 +19,10 @@ async function Prompter(program, requests) {
     envConfig.serverHost = 'https://api.forestadmin.com';
   }
 
+  if (process.env.FRONTEND_HOST) {
+    envConfig.frontendHost = process.env.FRONTEND_HOST;
+  }
+
```
Comment (SeyZ) [inline services/prompter.js]: Same here. We should keep this file untouched.

File: templates/app/app.js [hunk 3]
```diff
@@ -19,7 +19,7 @@ app.use(cookieParser());
 app.use(express.static(path.join(__dirname, 'public')));
 
 app.use(cors({
-  allowedOrigins: ['*.forestadmin.com'],
+  allowedOrigins: ['*.forestadmin.com', 'localhost:4200'],
```
Comment (SeyZ) [inline templates/app/app.js]: Something like: `allowedOrigins: [process.env.FOREST_HOSTNAME || '*.forestadmin,.com'],` ?

File: templates/app/env [hunk 4]
```diff
@@ -7,3 +7,4 @@ FOREST_ENV_SECRET=<%= forestEnvSecret %>
 <% if (devRenderingId) {%>DEV_RENDERING_ID=<%= devRenderingId %><% } %>
 <% if (dbSchema) {%>DATABASE_SCHEMA=<%= dbSchema %><% } %>
 <% if (port) {%>PORT=<%= port %><% } %>
+<% if (frontendHost) {%>FRONTEND_HOST=<%= frontendHost %><% } %>
```
Comment (SeyZ) [inline templates/app/env]: Same here. I prefer to set the environment variable in the dotenv manually or run my lumber back office application with: `FRONTEND_HOST=http://localhost:4200 npm start` if you don't want to set it into the .env

**Comment by arnaudbesnier** [comment]
Did I set you as a reviewer? It was just a draft to help me test/debug onboardings fastly.
If I were you, I would not lose time to review something that is not clear or finished.

That being said, we should find a "dev mode" for Lumber, it might be with a single option instead of multiple configuration variables.
Without it, the dev team won't be performant to test, debug or implement features.
(I did at least 20 generates command and it was painful to setup manually the variables in the generated projects).

**Pull request closed without merging**